Clarify SD & LB docs.

brndnmtthws opened the pull request:
This PR cleans up the SD & LB landing page, removes non-facts, and
clarifies the format for VIPs.

These changes are to address the frequent questions I'm seeing via
slack, email, github issues, and elsewhere.

brndnmtthws commented:
Also cc @pyronicide

pyronicide commented on 1.7/usage/service-discovery/index.md:
```diff
@@ -4,22 +4,16 @@ nav_title: Service Discovery
-If you're unable to use Mesos-DNS or VIPs, you can use a [load balanced solution][3].
+| Tool | Type | When should I use it? |
+|----------------------------|-------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| [VIPs][1] ([Minuteman][2]) | Minuteman is a layer 4 load balancer, which can be used for most TCP traffic for any Mesos task within a DC/OS cluster. | When you have internal-only TCP services which do not require layer 7 features. |
```
UDP as well.

brndnmtthws replied:
I thought minuteman was TCP-only?

pyronicide commented on 1.7/usage/service-discovery/virtual-ip-addresses.md:
```diff
@@ -24,11 +33,11 @@ To create a VIP:
 ## Assigning Multiple VIPs to Your App
 
-To assign multiple VIPs to your application, switch to JSON mode and add the additional VIPs to your `portDefinitions`. In this example, the additional VIP added is `"VIP_1": "111.2.1.23:5050"`:
+To assign multiple VIPs to your application, switch to JSON mode and add the additional VIPs to your `portDefinitions`. In this example, the additional VIP added is `"VIP_1": "192.168.0.1:5050"`:
```
Let's pick a better subnet here, maybe `172.0.0.0`?

brndnmtthws replied:
`172.0.0.0` isn't within the private range, but `172.16.0.0/12` is :)

Updated accordingly, and thanks for the suggestion.

joel-hamill commented:
LGTM. @pyronicide looks like an open question still for you.

brndnmtthws committed:
Clarify SD & LB docs.

This PR cleans up the SD & LB landing page, removes non-facts, and
clarifies the format for VIPs. Updated for 1.7 & 1.8.

These changes are to address the frequent questions I'm seeing via
slack, email, github issues, and elsewhere.

brndnmtthws commented:
@joel-hamill can you take another look please? I've updated this for 1.8.

joel-hamill commented on 1.8/usage/service-discovery/load-balancing-vips/virtual-ip-addresses.md:
```diff
@@ -4,6 +4,12 @@ menu_order: 10
 
+A named VIP contains 3 components:
+
+ * The private virtual IP address,
```
* Private virtual IP address
* Service port (the port which the service is available on)
* Service name

joel-hamill commented on 1.7/usage/service-discovery/virtual-ip-addresses.md:
```diff
@@ -3,9 +3,18 @@ post_title: Using Virtual IP Addresses
+DC/OS can map traffic from a single Virtual IP (VIP) and port to multiple IP addresses and ports. A VIP contains 2 components:
 
-You can assign a VIP to your application by using the DC/OS Marathon web interface. The values you enter in these fields are translated into the appropriate `portMapping` entry in your application definition. Toggle to `JSON mode` as you create your app to see and edit your application definition.
+ * The private virtual IP address, and
```
* Private virtual IP address
* Service port

joel-hamill commented:
@brndnmtthws left a couple formatting comments.

brndnmtthws commented:
@joel-hamill updated

pyronicide commented on 1.7/usage/service-discovery/virtual-ip-addresses.md:
```diff
@@ -3,9 +3,18 @@ post_title: Using Virtual IP Addresses
 
-You can assign a VIP to your application by using the DC/OS Marathon web interface. The values you enter in these fields are translated into the appropriate `portMapping` entry in your application definition. Toggle to `JSON mode` as you create your app to see and edit your application definition.
+ * Private virtual IP address
+ * Service port
```
You don't want to refer to it as a `service port`. How about just `port`?

brndnmtthws replied:
The terminology is confusing, because we refer to it as a service port in the Marathon docs (and it's conceptually the same). Regardless, I'm fine with dropping it.

brndnmtthws commented:
Updated again.

brndnmtthws commented:
I think we're good to go on this.
